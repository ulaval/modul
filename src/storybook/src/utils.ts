export const modulComponentsHierarchyRootSeparator: string = 'modul-components|';
export const modulWebsiteHierarchyRootSeparator: string = 'modul-website|';
