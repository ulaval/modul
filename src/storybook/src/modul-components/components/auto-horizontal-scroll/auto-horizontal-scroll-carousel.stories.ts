import { boolean, select } from '@storybook/addon-knobs';
import { MAutoHorizontalScrollGradientStyle } from '@ulaval/modul-components/dist/components/auto-horizontal-scroll/auto-horizontal-scroll';
import { AUTO_HORIZONTAL_SCROLL } from '@ulaval/modul-components/dist/components/component-names';
import { Enums } from '@ulaval/modul-components/dist/utils/enums/enums';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${AUTO_HORIZONTAL_SCROLL}/CarouselExample`,
    parameters: { fileName: __filename }
};

const getCarouselTemplate = (
    _horizontalScrollOffset: number,
    _nbItem: number,
    _displayButton: boolean
) => ({
    props: {
        leftGradientActive: {
            default: boolean('Prop left-gradient-active', true)
        },
        rightGradientActive: {
            default: boolean('Prop right-gradient-active', true)
        },
        previousButtonActive: {
            default: boolean('Prop previous-button-active', _displayButton)
        },
        nextButtonActive: {
            default: boolean('Prop next-button-active', _displayButton)
        },
        gradientStyle: {
            default: select(
                'Prop gradient-style',
                Enums.toValueArray(
                    MAutoHorizontalScrollGradientStyle
                ),
                MAutoHorizontalScrollGradientStyle.White
            )
        },
        displayHorizontalScrollbar: {
            default: boolean('Prop display-horizontal-scrollbar', true)
        }
    },
    data: () => ({
        horizontalScrollOffset: _horizontalScrollOffset,
        crouselItemSize: 200,
        crouselItemSpacing: 16,
        nbItem: _nbItem
    }),
    methods: {
        isActiveBox(horizontalScrollOffset: number, boxIndex: number): boolean {
            const _this: any = this;
            const boxFullWidth: number = _this.crouselItemSize + _this.crouselItemSpacing;
            return horizontalScrollOffset >= (((boxIndex - 1) * boxFullWidth) - 160)
                && horizontalScrollOffset < (((boxIndex - 1) * boxFullWidth) + boxFullWidth - 160);
        },
        onPreviousButtonClick(): void {
            const _this: any = this;
            _this.incrementHorizontalScrollOffset(-_this.crouselItemSize - _this.crouselItemSpacing);
        },
        onNextButtonClick(): void {
            const _this: any = this;
            _this.incrementHorizontalScrollOffset(_this.crouselItemSize + _this.crouselItemSpacing);
        },
        incrementHorizontalScrollOffset(increment: number): void {
            const _this: any = this;
            _this.horizontalScrollOffset = _this.horizontalScrollOffset + increment;
        }
    },
    template: `<${AUTO_HORIZONTAL_SCROLL}
        :min-width="crouselItemSize * nbItem + (nbItem + 1) * crouselItemSpacing + 'px'"
        :left-gradient-active="leftGradientActive"
        :right-gradient-active="rightGradientActive"
        :previous-button-active="previousButtonActive"
        :next-button-active="nextButtonActive"
        :horizontal-scroll-offset.sync="horizontalScrollOffset"
        :gradient-style="gradientStyle"
        :display-horizontal-scrollbar="displayHorizontalScrollbar"
        style="background: #f4f4f4;"
        @previous-button-click="onPreviousButtonClick()"
        @next-button-click="onNextButtonClick()"
    >
        <div style="display: flex; padding: 32px 16px;">
            <article
                v-for="index in nbItem"
                :key="index"
                :data-box-index="index"
                style="transition: background-color 0.3s ease; flex-shrink: 0; padding: 16px;"
                :style="{
                    marginRight: index < nbItem ? crouselItemSpacing + 'px' : undefined,
                    backgroundColor: isActiveBox(horizontalScrollOffset, index) ? '#ffc103': '#fff',
                    width: crouselItemSize + 'px',
                    height: crouselItemSize + 'px',
                }"
            >
                <p class="m-u--no-margin">Carousel item #{{ index }}</p>
            </article>
        </div>
    </${AUTO_HORIZONTAL_SCROLL}>`
});

export const Sandbox = () => getCarouselTemplate(1200, 20, false);

export const TwoItems = () => getCarouselTemplate(0, 2, false);
export const WithButton = () => getCarouselTemplate(120, 14, true);
