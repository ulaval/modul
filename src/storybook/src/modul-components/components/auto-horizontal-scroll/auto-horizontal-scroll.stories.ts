import { actions } from '@storybook/addon-actions';
import { boolean, select, text } from '@storybook/addon-knobs';
import { MAutoHorizontalScrollGradientStyle } from '@ulaval/modul-components/dist/components/auto-horizontal-scroll/auto-horizontal-scroll';
import { AUTO_HORIZONTAL_SCROLL } from '@ulaval/modul-components/dist/components/component-names';
import { Enums } from '@ulaval/modul-components/dist/utils/enums/enums';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${AUTO_HORIZONTAL_SCROLL}`,
    parameters: { fileName: __filename }
};

const getTemplate = (
    _previousButtonActive: boolean,
    _nextButtonActive: boolean,
    _leftGradientActive: boolean,
    _rightGradientActive: boolean,
    _previousButtonText: string = 'Previous',
    _nextButtonText: string = 'Next',
    _dragActive: boolean = true,
    _displayHorizontalScrollbar: boolean = true
) => ({
    props: {
        minWidth: {
            default: text('Prop min-width', '1900px')
        },
        dragActive: {
            default: boolean('Prop drag-active', _dragActive)
        },
        previousButtonActive: {
            default: boolean('Prop previous-button-active', _previousButtonActive)
        },
        nextButtonActive: {
            default: boolean('Prop next-button-active', _nextButtonActive)
        },
        previousButtonText: {
            default: text('Prop previous-button-text', _previousButtonText)
        },
        nextButtonText: {
            default: text('Prop next-button-text', _nextButtonText)
        },
        leftGradientActive: {
            default: boolean('Prop left-gradient-active', _leftGradientActive)
        },
        rightGradientActive: {
            default: boolean('Prop right-gradient-active', _rightGradientActive)
        },
        gradientStyle: {
            default: select(
                'Prop gradient-style',
                Enums.toValueArray(
                    MAutoHorizontalScrollGradientStyle
                ),
                MAutoHorizontalScrollGradientStyle.White
            )
        },
        displayHorizontalScrollbar: {
            default: boolean('Prop display-horizontal-scrollbar', _displayHorizontalScrollbar)
        },
        slotDefault: {
            default: boolean('Slot default', true)
        }
    },
    methods: actions('emitPreviousButtonClick', 'emitNextButtonClick'),
    data: () => ({
        horizontalScrollOffset: 60
    }),
    template: `<${AUTO_HORIZONTAL_SCROLL}
        :min-width="minWidth"
        :drag-active="dragActive"
        :left-gradient-active="leftGradientActive"
        :right-gradient-active="rightGradientActive"
        :previous-button-active="previousButtonActive"
        :next-button-active="nextButtonActive"
        :previous-button-text="previousButtonText"
        :next-button-text="nextButtonText"
        :gradient-style="gradientStyle"
        :horizontal-scroll-offset.sync="horizontalScrollOffset"
        :display-horizontal-scrollbar="displayHorizontalScrollbar"
        @previous-button-click="emitPreviousButtonClick"
        @next-button-click="emitNextButtonClick"
    >
        <div v-if="slotDefault"
            style="background: #1f1f1f; padding: 32px; color: #fff; width: 100%;"
        >
            <p class="m-u--no-margin">Adjust the width of your browser to <strong>{{ parseInt(minWidth, 10) - 80 }}px</strong> to be able to test the component.</p>
            <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
            <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
        </div>
    </${AUTO_HORIZONTAL_SCROLL}>`
});

export const Sandbox = () => getTemplate(true, true, true, true);

export const PreviousButtonActive = () => getTemplate(true, false, false, false);
export const NextButtonActive = () => getTemplate(false, true, false, false);

export const LeftGradientActive = () => getTemplate(false, false, true, false);
export const RightGradientActive = () => getTemplate(false, false, false, true);

export const PreviousButtonText = () => getTemplate(true, false, false, false, 'Custom Previous');
export const NextButtonText = () => getTemplate(false, true, false, false, 'Previous', 'Custom Next');

export const DragDisabled = () => getTemplate(true, true, false, false, 'Previous', 'Next', false);
export const HideHorizontalScrollbar = () => getTemplate(true, true, false, false, 'Previous', 'Next', true, false);


