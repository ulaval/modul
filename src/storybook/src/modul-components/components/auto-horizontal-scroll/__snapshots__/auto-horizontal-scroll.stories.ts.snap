// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots modul-components|m-auto-horizontal-scroll Drag Disabled 1`] = `
<div
  class="m-auto-horizontal-scroll m--is-gradient-background-style-white"
  style="position: relative;"
>
  <!---->
   
  <!---->
   
  <div
    class="m-auto-horizontal-scroll__body"
  >
    <div
      class="m-auto-horizontal-scroll__body-content"
      style="min-width: 1900px;"
    >
      <div
        style="background: rgb(31, 31, 31); padding: 32px; color: rgb(255, 255, 255); width: 100%;"
      >
        <p
          class="m-u--no-margin"
        >
          Adjust the width of your browser to 
          <strong>
            1820px
          </strong>
           to be able to test the component.
        </p>
         
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
         
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
      </div>
    </div>
  </div>
  <div
    class="resize-sensor"
    dir="ltr"
    style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
  >
    <div
      class="resize-sensor-expand"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s;"
      />
    </div>
    <div
      class="resize-sensor-shrink"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 200%; height: 200%;"
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots modul-components|m-auto-horizontal-scroll Hide Horizontal Scrollbar 1`] = `
<div
  class="m-auto-horizontal-scroll m--is-gradient-background-style-white"
  style="position: relative;"
>
  <!---->
   
  <!---->
   
  <div
    class="m-auto-horizontal-scroll__body"
  >
    <div
      class="m-auto-horizontal-scroll__body-content"
      style="min-width: 1900px;"
    >
      <div
        style="background: rgb(31, 31, 31); padding: 32px; color: rgb(255, 255, 255); width: 100%;"
      >
        <p
          class="m-u--no-margin"
        >
          Adjust the width of your browser to 
          <strong>
            1820px
          </strong>
           to be able to test the component.
        </p>
         
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
         
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
      </div>
    </div>
  </div>
  <div
    class="resize-sensor"
    dir="ltr"
    style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
  >
    <div
      class="resize-sensor-expand"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s;"
      />
    </div>
    <div
      class="resize-sensor-shrink"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 200%; height: 200%;"
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots modul-components|m-auto-horizontal-scroll Left Gradient Active 1`] = `
<div
  class="m-auto-horizontal-scroll m--is-gradient-background-style-white"
  style="position: relative;"
>
  <!---->
   
  <!---->
   
  <div
    class="m-auto-horizontal-scroll__body"
  >
    <div
      class="m-auto-horizontal-scroll__body-content"
      style="min-width: 1900px;"
    >
      <div
        style="background: rgb(31, 31, 31); padding: 32px; color: rgb(255, 255, 255); width: 100%;"
      >
        <p
          class="m-u--no-margin"
        >
          Adjust the width of your browser to 
          <strong>
            1820px
          </strong>
           to be able to test the component.
        </p>
         
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
         
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
      </div>
    </div>
  </div>
  <div
    class="resize-sensor"
    dir="ltr"
    style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
  >
    <div
      class="resize-sensor-expand"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s;"
      />
    </div>
    <div
      class="resize-sensor-shrink"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 200%; height: 200%;"
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots modul-components|m-auto-horizontal-scroll Next Button Active 1`] = `
<div
  class="m-auto-horizontal-scroll m--is-gradient-background-style-white"
  style="position: relative;"
>
  <!---->
   
  <!---->
   
  <div
    class="m-auto-horizontal-scroll__body"
  >
    <div
      class="m-auto-horizontal-scroll__body-content"
      style="min-width: 1900px;"
    >
      <div
        style="background: rgb(31, 31, 31); padding: 32px; color: rgb(255, 255, 255); width: 100%;"
      >
        <p
          class="m-u--no-margin"
        >
          Adjust the width of your browser to 
          <strong>
            1820px
          </strong>
           to be able to test the component.
        </p>
         
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
         
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
      </div>
    </div>
  </div>
  <div
    class="resize-sensor"
    dir="ltr"
    style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
  >
    <div
      class="resize-sensor-expand"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s;"
      />
    </div>
    <div
      class="resize-sensor-shrink"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 200%; height: 200%;"
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots modul-components|m-auto-horizontal-scroll Next Button Text 1`] = `
<div
  class="m-auto-horizontal-scroll m--is-gradient-background-style-white"
  style="position: relative;"
>
  <!---->
   
  <!---->
   
  <div
    class="m-auto-horizontal-scroll__body"
  >
    <div
      class="m-auto-horizontal-scroll__body-content"
      style="min-width: 1900px;"
    >
      <div
        style="background: rgb(31, 31, 31); padding: 32px; color: rgb(255, 255, 255); width: 100%;"
      >
        <p
          class="m-u--no-margin"
        >
          Adjust the width of your browser to 
          <strong>
            1820px
          </strong>
           to be able to test the component.
        </p>
         
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
         
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
      </div>
    </div>
  </div>
  <div
    class="resize-sensor"
    dir="ltr"
    style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
  >
    <div
      class="resize-sensor-expand"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s;"
      />
    </div>
    <div
      class="resize-sensor-shrink"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 200%; height: 200%;"
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots modul-components|m-auto-horizontal-scroll Previous Button Active 1`] = `
<div
  class="m-auto-horizontal-scroll m--is-gradient-background-style-white"
  style="position: relative;"
>
  <!---->
   
  <!---->
   
  <div
    class="m-auto-horizontal-scroll__body"
  >
    <div
      class="m-auto-horizontal-scroll__body-content"
      style="min-width: 1900px;"
    >
      <div
        style="background: rgb(31, 31, 31); padding: 32px; color: rgb(255, 255, 255); width: 100%;"
      >
        <p
          class="m-u--no-margin"
        >
          Adjust the width of your browser to 
          <strong>
            1820px
          </strong>
           to be able to test the component.
        </p>
         
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
         
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
      </div>
    </div>
  </div>
  <div
    class="resize-sensor"
    dir="ltr"
    style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
  >
    <div
      class="resize-sensor-expand"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s;"
      />
    </div>
    <div
      class="resize-sensor-shrink"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 200%; height: 200%;"
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots modul-components|m-auto-horizontal-scroll Previous Button Text 1`] = `
<div
  class="m-auto-horizontal-scroll m--is-gradient-background-style-white"
  style="position: relative;"
>
  <!---->
   
  <!---->
   
  <div
    class="m-auto-horizontal-scroll__body"
  >
    <div
      class="m-auto-horizontal-scroll__body-content"
      style="min-width: 1900px;"
    >
      <div
        style="background: rgb(31, 31, 31); padding: 32px; color: rgb(255, 255, 255); width: 100%;"
      >
        <p
          class="m-u--no-margin"
        >
          Adjust the width of your browser to 
          <strong>
            1820px
          </strong>
           to be able to test the component.
        </p>
         
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
         
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
      </div>
    </div>
  </div>
  <div
    class="resize-sensor"
    dir="ltr"
    style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
  >
    <div
      class="resize-sensor-expand"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s;"
      />
    </div>
    <div
      class="resize-sensor-shrink"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 200%; height: 200%;"
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots modul-components|m-auto-horizontal-scroll Right Gradient Active 1`] = `
<div
  class="m-auto-horizontal-scroll m--is-gradient-background-style-white"
  style="position: relative;"
>
  <!---->
   
  <!---->
   
  <div
    class="m-auto-horizontal-scroll__body"
  >
    <div
      class="m-auto-horizontal-scroll__body-content"
      style="min-width: 1900px;"
    >
      <div
        style="background: rgb(31, 31, 31); padding: 32px; color: rgb(255, 255, 255); width: 100%;"
      >
        <p
          class="m-u--no-margin"
        >
          Adjust the width of your browser to 
          <strong>
            1820px
          </strong>
           to be able to test the component.
        </p>
         
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
         
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
      </div>
    </div>
  </div>
  <div
    class="resize-sensor"
    dir="ltr"
    style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
  >
    <div
      class="resize-sensor-expand"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s;"
      />
    </div>
    <div
      class="resize-sensor-shrink"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 200%; height: 200%;"
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots modul-components|m-auto-horizontal-scroll Sandbox 1`] = `
<div
  class="m-auto-horizontal-scroll m--is-gradient-background-style-white"
  style="position: relative;"
>
  <!---->
   
  <!---->
   
  <div
    class="m-auto-horizontal-scroll__body"
  >
    <div
      class="m-auto-horizontal-scroll__body-content"
      style="min-width: 1900px;"
    >
      <div
        style="background: rgb(31, 31, 31); padding: 32px; color: rgb(255, 255, 255); width: 100%;"
      >
        <p
          class="m-u--no-margin"
        >
          Adjust the width of your browser to 
          <strong>
            1820px
          </strong>
           to be able to test the component.
        </p>
         
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
         
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
      </div>
    </div>
  </div>
  <div
    class="resize-sensor"
    dir="ltr"
    style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
  >
    <div
      class="resize-sensor-expand"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s;"
      />
    </div>
    <div
      class="resize-sensor-shrink"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 200%; height: 200%;"
      />
    </div>
  </div>
</div>
`;
